### DISCUSSION TITLE
Folder containing only hidden files is displayed as empty only after opening

### DISCUSSION THREAD

**Discussion by dxrcy** [nvim-tree/nvim-tree.lua#2796] [General]
Given the file `foo/.bar` in the working directory, the folder `foo` will display with the normal, non-empty, folder icon.
Then when it is opened (revealing no children, because dotfiles are hidden), the icon will switch to the empty (open) folder icon.
When closed again, the folder will show the empty (closed) folder icon.

Is this intentional? Or perhaps a limitation of the way directories are read?

I feel like it should either:
**a)** Show the empty folder icon before being opened
or
**b)** Show as non-empty, even if all children are hidden or otherwise filtered (perhaps with some indicator)

This relates to #2225 (especially considering option '**b**').
Also possibly related is #2496.

# Example

Directory structure:
![A `tree` output of a directory with children `foo/` and `foo/.bar`](https://github.com/nvim-tree/nvim-tree.lua/assets/44690813/14891fad-0a93-4b39-bc6f-24775acbba8c)

Before opening:
![Folder with non-empty closed icon](https://github.com/nvim-tree/nvim-tree.lua/assets/44690813/47a63cf8-6f5a-4472-940d-a23fe84a0562)
After opening:
![Folder with empty opened icon](https://github.com/nvim-tree/nvim-tree.lua/assets/44690813/c5f3366c-03e4-4afb-911a-b4442c372357)
After closing:
![Folder with empty closed icon](https://github.com/nvim-tree/nvim-tree.lua/assets/44690813/ef93862a-2395-4553-ae31-4db898b49d29)

**Comment by alex-courtis** [comment]
Reproduced. Genuinely empty folders show the closed empty folder icon before they are opened.

When populating each folder the contents are not enumerated, only whether it contains something. That's an intentional performance decision.

I think b is the practical choice here - it's consistent and correct.

Do you feel like a pull request @dxrcy ?

**Reply by dxrcy** [reply]
> That's an intentional performance decision.

I thought it may have been.

> Do you feel like a pull request @dxrcy ?

Yep I'll give it a look 👍
